java/src/main/java/ai/rapids/cudf/ColumnView.java

```diff
@@ -1392,42 +1392,91 @@ public final ColumnVector rollingWindow(RollingAggregation op, WindowOptions opt
    * This is just a convenience method for an inclusive scan with a SUM aggregation.
    */
   public final ColumnVector prefixSum() {
-    return scan(Aggregation.sum());
+    return scan(this, Aggregation.sum());
   }
 
   /**
    * Computes a scan for a column. This is very similar to a running window on the column.
-   * @param aggregation the aggregation to perform
+   * @param input native id of a ColumnVector or ColumnView, a 0 (null column) is ok for
+   *              rank and dense rank aggregations.
+   * @param aggregation the aggregation to perform.
    * @param scanType should the scan be inclusive, include the current row, or exclusive.
    * @param nullPolicy how should nulls be treated. Note that some aggregations also include a
    *                   null policy too. Currently none of those aggregations are supported so
    *                   it is undefined how they would interact with each other.
    */
-  public final ColumnVector scan(Aggregation aggregation, ScanType scanType, NullPolicy nullPolicy) {
+  private static final ColumnVector scan(long input, Aggregation aggregation,
+      ScanType scanType, NullPolicy nullPolicy) {
     long nativeId = aggregation.createNativeInstance();
     try {
-      return new ColumnVector(scan(getNativeView(), nativeId,
+      return new ColumnVector(scan(input, nativeId,
           scanType.isInclusive, nullPolicy.includeNulls));
     } finally {
       Aggregation.close(nativeId);
     }
   }
 
```

Comment:
You deleted all of the non static API versions of scan except for prefixSum. This makes this a breaking change. Could you please add them back similar to how you did prefix sum.

Reply:
I realize that because of name collisions that both of them cannot be scan, but from an API perspective, the majority of the aggregations require a column so I would prefer to see that be the scan API and the other one be a no column scan or something like that.  Also generally static APIs that return a ColumnVector we have put in ColumnVector instead of ColumnView.